src/main/java/org/spongepowered/api/text/TextArgument.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.text;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+
+/**
+ * Represents a function that will be triggered inside {@link Text.Placeholder}
+ * to calculate the replacement for the {@link Text.Placeholder}.
+ */
+public interface TextArgument {
+
+    /**
+     * Resolve the result of TextArgument. If this instance is the main
```

Comment:
I could not understand what the javadocs said, neither what you just said. The javadocs state to me a very clear condition:

> If this instance is the "main" text argument _and_ this method returns Optional.absent(), then the wrapping TextPlaceholder will not be replaced.

To me, that doesn't actually describe how the method behaves, it seems to me that it describes an edge case.